node/chainSimulator/components/testOnlyProcessingNode.go

```diff
@@ -411,7 +428,48 @@ func (node *testOnlyProcessingNode) SetState(address []byte, keyValueMap map[str
 		}
 	}
 
-	err = accountsAdapter.SaveAccount(account)
+	return nil
+}
+
+// SetStateForAddress will set the state for the give address
+func (node *testOnlyProcessingNode) SetStateForAddress(address []byte, addressState *dtos.AddressState) error {
+	userAccount, err := node.getUserAccount(address)
```

Comment:
what happens if the address is not in the shard maintained by this testOnlyProcessingNode instance?

Reply:
same here